### ISSUE TITLE
Prevent KB4577586 from removing Adobe Flash Player

### ISSUE THREAD

**Issue by jeffythedragonslayer** [pakastin/open-source-flash#140]
I want to keep my windows 10 up to date, but don't want the update KB4577586 to install that will remove Flash.  I can see it pending download, along with other non deleterious updates.  The 'Download' button usually downloads all of these non-optional updates though.  Does anyone know a way to blacklist certain updates from installing?

**Comment by Brian151** [comment]
i think it falls under the realm of more generalized hacks/mods to the windows OS, itself...
since, yanno, in the name of our "security" and the system's "stability" windows 10 is only slightly less locked down than iOS, really... 
[with the added bonus the updater can mess-up and delete files/settings/drivers or brick the whole machine]

**Comment by jeffythedragonslayer** [comment]
Yeah.  I have definitely heard of people uninstalling windows updates, but this update itself is an uninstallation.  Don't know if the double negative of letting all the pending updates go through and then uninstalling KB4577586 will put Flash back.

**Comment by jeffythedragonslayer** [comment]
I tried letting the "update" install on another machine and then tried `wusa /uninstall /KB:4577586` but got this error:

![image](https://user-images.githubusercontent.com/13253146/108615292-b317b800-73d0-11eb-827e-752cc2290e82.png)

oh really?

**Comment by Brian151** [comment]
i probably shouldn't say this on a public forum, but...
wonder when this excuse will be recycled for making installing from the store non-optional?
there's already a setting for this, and it's "recommended".

basically the same "your computer NEEDS this" was used to lock-down the updater so tightly to begin with.

**Comment by JPcode05** [comment]
I think maybe an option would be to create a third-party patch that allows you to uninstall KB4577586. The patch that uninstalls the patch, basically.

**Comment by jeffythedragonslayer** [comment]
@JPcode05 well if you think that can be done I suppose we could create a new issue?

**Comment by kristibektashi** [comment]
Can't you theoretically use the ActiveX installer of Flash Player with Windows 7 compatibility settings to reinstall Flash Player?
